src/nvim/tui/tui.c

```diff
@@ -1973,7 +1973,23 @@ static void flush_buf(UI *ui)
   uv_buf_t *bufp = &bufs[0];
   TUIData *data = ui->data;
 
-  if (data->bufpos <= 0 && data->busy == data->is_invisible) {
+  // The content of the output for each condition is shown in the following
+  // table. Therefore, if data->bufpos == 0 and N/A or invis + norm, there is
+  // no need to output it.
+  //
+  //                         | is_invisible | !is_invisible
+  // ------+-----------------+--------------+---------------
+  // busy  | want_invisible  |     N/A      |    invis
+  //       | !want_invisible |     N/A      |    invis
+  // ------+-----------------+--------------+---------------
+  // !busy | want_invisible  |     N/A      |    invis
+  //       | !want_invisible |     norm     | invis + norm
+  // ------+-----------------+--------------+---------------
+  //
+  if (data->bufpos <= 0
+      && ((data->busy && data->is_invisible)
+          || (data->is_invisible && !data->busy && data->want_invisible)
+          || (!data->is_invisible && !data->busy && !data->want_invisible))) {
```

Comment:
if I nitpick I would keep the order consistent between the different condition, and ideally with the table:
(data->busy / is_invisible / want_invisible

Reply:
It is corrected to the processing order ( is_invisible / data->busy / want_invisible. I think `busy` and  `want_invisible` are so closely related that it's better to have them next to each other. The table has not been modified because the number of columns exceeds 80 characters. Sure, it's a little hard to see, but if you read it vertically, it's not hard to read, so how about this?